File: src/primus/pawel/Exercise_174.java
package primus.pawel;

// Word values
// 7kyu
// https://www.codewars.com/kata/598d91785d4ce3ec4f000018/train/java
// 2021-07-09 T:19:58:21
class Exercise_174 {

    public static int[] nameValue(String[] arr) {
        int[] result = new int[arr.length];
        int valueOfCurrentLoop = 0;
        char currentChar;

        for (int i = 0; i < arr.length; i++) {

            for (int j = 0; j < arr[i].length(); j++) {
                currentChar = arr[i].charAt(j);
                if (currentChar != ' ') {
                    valueOfCurrentLoop += (currentChar - 96);
                }

            }
            result[i] = valueOfCurrentLoop*(i+1);
            valueOfCurrentLoop = 0;
        }

        return result;
    }

    public static void main(String[] args) {

        System.out.println(nameValue(new String[]{"codewars", "abc", "xyz"}));
    }
}

/*   public static int [] nameValue(String [] arr){
        int[] result = new int[arr.length];
        for (int i = 0; i < arr.length; i++){
            result[i] = arr[i].chars().filter(e -> e != ' ').map(e -> e - 96).sum() * (i+1);
        }
        return result;
    }
    */

File: src/JS/Exercise_174.js
// Substring fun
// 7kyu
// https://www.codewars.com/kata/565b112d09c1adfdd500019c/train/javascript
// 2022-07-21 T:07:32:41

function nthChar(words) {
    var result = '';

    for (let i = 0; i < words.length; i++) {
        result += words[i][i]
    }
    return result;
}

/*function nthChar(words){
 return words.map((str, i) => str.charAt(i)).join("");
}*/